dmr/src/main/java/org/jboss/hal/dmr/dispatch/Dispatcher.java

```diff
@@ -77,29 +77,33 @@
 public class Dispatcher implements RecordingHandler {
 
     @FunctionalInterface
-    public interface SuccessCallback<T> {
+    public interface OnSuccess {
```

Comment:
I like the idea to reuse the functions from `java.util.function` (like you did in `BrowseContentElement.java` or  `ApplyPatchWizard.java`). So I think the `OnSuccess` and `CompositeCallback` can be replace with `Consumer<ModelNode>` and `Consumer<CompositeResult>`.

Reply:
yep, I also start doing that in this PR but it changes a lot too and, you should not that pretty soon you will just remove this callback, you should only use the RX container, and lambdas applied using some RX operator, so whatever changes are applied to this type it will disappear. So what I was doing is to align the API with RX which requires a clear success/error API. I think now it is more important to join the failure/error callbacks, if you really need to handle it in some place you can always do `ex instanceof DispatcherFailureException` or `ex instanceof DispatcherErrorException`. And after that all this method should look like `void xxx(_input_args_, OnSuccess, OnError)` which is trivial to apply RX as `Single<ModelNode> xxx(_input_args_)`. At this point, the OnSuccess and OnFailure can be removed.

Reply:
Going to replace the success callbacks with `Consumer<T>`